be/src/storage/lake/delta_writer.cpp

```diff
@@ -0,0 +1,238 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#include "storage/lake/delta_writer.h"
+
+#include "column/chunk.h"
+#include "column/column.h"
+#include "gutil/strings/util.h"
+#include "runtime/current_thread.h"
+#include "runtime/exec_env.h"
+#include "runtime/mem_tracker.h"
+#include "storage/lake/tablet.h"
+#include "storage/lake/tablet_manager.h"
+#include "storage/lake/tablet_writer.h"
+#include "storage/lake/txn_log.h"
+#include "storage/memtable.h"
+#include "storage/memtable_flush_executor.h"
+#include "storage/memtable_sink.h"
+#include "storage/storage_engine.h"
+
+namespace starrocks::lake {
+
+using Chunk = starrocks::vectorized::Chunk;
+using Column = starrocks::vectorized::Column;
+using MemTable = starrocks::vectorized::MemTable;
+using MemTableSink = starrocks::vectorized::MemTableSink;
+
+class TabletWriterSink : public MemTableSink {
+public:
+    explicit TabletWriterSink(TabletWriter* w) : _writer(w) {}
+
+    ~TabletWriterSink() override = default;
+
+    DISALLOW_COPY_AND_MOVE(TabletWriterSink);
+
+    Status flush_chunk(const Chunk& chunk) override {
+        RETURN_IF_ERROR(_writer->write(chunk));
+        return _writer->flush();
+    }
+
+    Status flush_chunk_with_deletes(const Chunk& /*upserts*/, const Column& /*deletes*/) override {
+        return Status::NotSupported("TabletWriterSink::flush_chunk_with_deletes");
+    }
+
+private:
+    TabletWriter* _writer;
```

Comment:
Why not use `RowsetWriter` as name?

Reply:
IMHO, it would be more natural to have a `TabletWriter` along with a `TabletReader`.

Reply:
Just write one rowset in writer, but read all rowsets in reader.